### ISSUE TITLE
Keras docs source code links point to 404

### ISSUE THREAD

**Issue by grofte** [tensorflow/tensorflow#61429]
Example:

if I go here https://www.tensorflow.org/api_docs/python/tf/keras/Model

and click source code I get to a 404 here https://github.com/keras-team/keras/tree/v2.13.1/keras/src/engine/training.py#L70-L3991

And it's the same no matter what version of the API docs I click from.  
Speaking of, if you go to https://www.tensorflow.org/versions then the 2.12 and 2.11 links don't point to those versions of the API docs. They go to the current versions. Unless you use the links in the left side bar - those are still okay. Except that there's no link to the current version - which is 2.13.

**Comment by SuryanarayanaY** [comment]
Hi @grofte ,

This is already brought to the Docs team notice and they are working on fixing it.

However if you are looking for source code of https://github.com/keras-team/keras/tree/v2.13.1/keras/src/engine/training.py#L70-L3991 then remove `src/` from the URL address and it works fine.This is just a temporary workaround. The problem exists in TF2.13 due to some internal movements.

If you want to access older versions you can still access without any problem. For example please look for TF2.12v API docs [here](https://www.tensorflow.org/versions/r2.12/api_docs/python/tf).

**Comment by grofte** [comment]
Okay, I didn't find the `docs-bug` label when I was searching before creating this issue. 

W.r.t. the older API documentation I am aware that it still exists. The links from vers. 2.10 and down work fine so I could open them and edit the address bar. But that doesn't change that both the 2.11 and 2.12 links point to https://www.tensorflow.org/api_docs/python/tf  
The release note links are fine though. I don't know why those two are wrong.

**Comment by SuryanarayanaY** [comment]
@grofte ,

I am not sure I am getting your problem wrt 2.12 avd 2.11 versions. I can see the links are working fine for these versions.

https://www.tensorflow.org/versions/r2.12/api_docs/python/tf

https://www.tensorflow.org/versions/r2.11/api_docs/python/tf

**Comment by grofte** [comment]
![v2 11 and v2 12](https://github.com/tensorflow/tensorflow/assets/7976840/906263ab-3f8b-4151-9f16-c321f2c57380)
![v2 10](https://github.com/tensorflow/tensorflow/assets/7976840/5a1982ee-b9ba-42d5-b376-19be230e3919)

On the versions page https://www.tensorflow.org/versions

**Comment by SuryanarayanaY** [comment]
Hi @grofte ,

Please check this.

<img width="1456" alt="Screenshot 2023-08-01 at 10 19 36 AM" src="https://github.com/tensorflow/tensorflow/assets/116063290/5cb513b0-4d1d-44d1-b2b3-10dc07cc4c0e">

**Comment by grofte** [comment]
It's still the same. The link right next to where it says "release notes" is wrong for r2.12 and r2.11

**Comment by SuryanarayanaY** [comment]
Hi @grofte ,

I have checked all the 3 nos of marked release notes (2.13, 2.12 and 2.11 respectively) in the below snapshot and its working fine and pointing to Release notes of respective version.


<img width="1456" alt="Screenshot 2023-08-01 at 10 19 36 AM" src="https://github.com/tensorflow/tensorflow/assets/116063290/fb12ad64-5a59-4450-855a-21c77c62b6e7">

**Comment by grofte** [comment]
I give up.

**Comment by SuryanarayanaY** [comment]
I noticed that the release note links of 2.14v and 2.13v both are pointing to 2.12v release notes.Submitting an internal fix for this.

**Comment by SuryanarayanaY** [comment]
@grofte ,

The fix has been merged and now I can see correct links for each version release notes. Could you please verify and close the issue now?

Thank you!

**Comment by grofte** [comment]
If I click r2.13, r2.12, or r2.11 they still lead to the 2.14 API documents.

If I click r2.11 it leads to 

https://www.tensorflow.org/api_docs/python/tf

but it should lead to

https://www.tensorflow.org/versions/r2.11/api_docs/python/tf

**Comment by SuryanarayanaY** [comment]
Hi @grofte ,

Are you referring to the below link marked ?

<img width="1331" alt="Screenshot 2023-10-18 at 8 13 37 PM" src="https://github.com/tensorflow/tensorflow/assets/116063290/e2022620-4f4f-4fe6-878e-cfaf10ff0c23">



 It is redirecting to Tf2.13.v link attached below.

https://www.tensorflow.org/versions/r2.13/api_docs/python/tf

Please confirm.

**Comment by SuryanarayanaY** [comment]
Hi @grofte ,

The issue being addressed in #62389. I have submitted a fix for this today. Will update you once it merged. Thanks!

**Comment by tilakrayal** [comment]
@grofte,
There was an internal issue raised for tracking the same issue and it got resolved now. Could you please try to access any Keras in the Tensorflow.org page, now it is able to re-direct to correct source code. Please check and confirm if it is working in your case.
https://www.tensorflow.org/api_docs/python/tf/keras/Model
https://github.com/keras-team/keras/blob/v3.3.3/keras/src/models/model.py#L32-L548

Thank you!

**Comment by grofte** [comment]
> @grofte, There was an internal issue raised for tracking the same issue and it got resolved now. Could you please try to access any Keras in the Tensorflow.org page, now it is able to re-direct to correct source code instead of 404 error page. Please check and confirm if it is working in your case. https://www.tensorflow.org/api_docs/python/tf/keras/Model https://github.com/keras-team/keras/blob/v3.3.3/keras/src/models/model.py#L32-L548
> 
> Thank you!

Yes, I seems to work for every point (except 2.16 but you have a link to 2.16.1 just over it that works).

**Comment by tilakrayal** [comment]
@grofte,
Could you please confirm which link you are mentioning either the keras documents or the tf core versions documents? Thank you!

**Comment by grofte** [comment]
https://www.tensorflow.org/api/r2.16

**Comment by tilakrayal** [comment]
@grofte,
For every stale version, the official API page will be as below. You can try on any of the links which are shown in the screenshot which will redirect to the correct page.
https://www.tensorflow.org/api_docs/python/tf

![image](https://github.com/tensorflow/tensorflow/assets/81610181/4cfe162c-74bf-496d-a924-71ff341ee2e3)

**Comment by grofte** [comment]
You were the one asking me whether there were any dead or misdirecting links. There's just one of them left as far as I can see.

![tensorflow2 16](https://github.com/tensorflow/tensorflow/assets/7976840/8410c181-1364-4805-91d5-1e203adc93b8)

**Comment by tilakrayal** [comment]
@grofte,
Yeah, As you mentioned that **2.16**(https://www.tensorflow.org/api/r2.16) link is re-directing to the **404** error page.  At present the tensorflow **v2.16.1(stable)**
https://www.tensorflow.org/api_docs/python/tf  is the stable version, So the 2.16.1(stable) is an active link.

AFAIK, once the upcoming version **2.17** is released, then the **2.16** link will get activated and the stable one becomes the new version(2.17 in this case) link. Thank you!
